x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichCache.java

```diff
@@ -52,6 +54,25 @@ public final class EnrichCache {
         this.cache = CacheBuilder.<CacheKey, List<Map<?, ?>>>builder().setMaximumWeight(maxSize).build();
     }
 
+    public void putIfAbsent(
```

Comment:
The semantics of this are different enough from [`Map#putIfAbsent(key, value)`](https://docs.oracle.com/en%2Fjava%2Fjavase%2F21%2Fdocs%2Fapi%2F%2F/java.base/java/util/Map.html#putIfAbsent(K,V)) that I think a different name is probably warranted -- it's a bit closer to `computeIfAbsent`, but I think a new different unique-to-this-class name might do a better job of indicating just how different the semantics of this are than anything else we've seen before. I might be overthinking this, of course. What do you think?

Regardless of whether the name changes, we'll probably want a docstring.

Reply:
Good point. Actually it looks like GeoIPCache's `putIfAbsent` (which I had copied this from) probably ought to be `computeIfAbsent` as well, since it also takes a function. And this one is slightly different, in that not only is it computing the value from a function -- it potentially does that asynchronously and notifies a listener on completion rather than returning the value. Maybe `computeAsyncIfAbsent`? I'll change it to `computeIfAbsent` for now. Let me know what you think.